app/models/changeset.rb

```diff
@@ -84,13 +84,32 @@ def find_comparison
 
   def find_pull_requests
     numbers = commits.map(&:pull_request_number).compact
-    numbers.map { |num| PullRequest.find(repo, num) }.compact
+    numbers.map { |num| PullRequest.find(repo, num) }.
+      compact.
+      concat(find_pull_request_for_branch)
   end
 
   def cache_key
     [self.class, repo, previous_commit, commit].join('-')
   end
 
+  def find_pull_request_for_branch
+    # Bail if commit looks like a commit SHA
+    return [] if skip_pull_request_fetch_from_api?
+    org = repo.split("/", 2).first
+    GITHUB.pull_requests(repo, head: "#{org}:#{commit}")
+  rescue Octokit::Error, Faraday::ConnectionFailed => e
+    Rails.logger.warn "Failed fetching pull requests for branch #{commit}:\n#{e}"
+    []
+  end
+
+  def skip_pull_request_fetch_from_api?
```

Comment:
`def looks_like_a_branch?`
and use `&&` to join things and avoid all the if/returns

Reply:
I'll use the `&&`
Are you sure about the name change because `master` (`staging`, ...) are branches, so `looks_like_a_branch?` does not really reflect the code.

Reply:
ah yes ... maybe `pr_branch?`

Reply:
Maybe we should not put these branches there at all, maybe some users use PRs to move between `staging` to `production`...

Reply:
`master` (and `develop`) won't make that request anyway cause they are converted to SHA, and it's not a big deal to make another request that will come back empty

Reply:
sounds reasonable